src/codec.rs

```diff
@@ -103,6 +103,11 @@ pub trait Encode {
     fn get_encoded(&self) -> Vec<u8> {
         self.get_encoded_with_param(&())
     }
+
+    /// Returns a hint indicating how many bytes will be required to encode this value.
```

Comment:
Justifying `Option<usize>` over `usize` as the return type will be useful IMO.

```suggestion
    /// Returns a hint indicating how many bytes will be required to encode this value.
    /// No hint is returned by default.
```